test/spec_rack_nested_params.rb

```diff
@@ -37,10 +37,10 @@ def middleware
     req.put?.must_equal true
   end
 
-  specify "should make first boolean have precedence even after request already parsed" do
+  specify "should make last boolean have precedence even after request already parsed" do
     app = Rack::MethodOverride.new(middleware)
     req = app.call(form_post("foo=1&foo=0")).last
-    req.POST.must_equal({"foo" => '1'})
+    req.POST.must_equal({"foo" => "0"})
```

Comment:
this seems like a subtle and potentially dangerous change - why are changing our the test cases to accommodate?

Reply:
This behavior will change with 2.0. See [this issue](https://github.com/rack/rack-contrib/issues/92) (and maybe [this too](https://github.com/rack/rack-contrib/pull/137)).  A deprecation warning was added to [v1.8.0](https://github.com/rack/rack-contrib/releases/tag/v1.8.0).

Reply:
@sshaw makes sense - thanks for the documentation